src/s390x/zipl.rs

```diff
@@ -29,22 +39,159 @@ pub fn chreipl<P: AsRef<Path>>(dev: P) -> Result<()> {
     Ok(())
 }
 
-/// Runs `zipl` based on Ignition and BLS configuration in `boot`.
-pub fn zipl<P: AsRef<Path>>(boot: P) -> Result<()> {
+fn secure_execution_is_enabled() -> Result<bool> {
+    let sysfs_flag = "/sys/firmware/uv/prot_virt_guest";
+    match File::open(sysfs_flag) {
+        Err(e) if e.kind() == std::io::ErrorKind::NotFound => Ok(false),
+        Err(e) => Err(e).with_context(|| format!("reading {}", sysfs_flag)),
+        Ok(mut f) => {
+            let mut buffer = String::new();
+            f.read_to_string(&mut buffer)?;
+            Ok(buffer.trim() == "1")
+        }
+    }
+}
+
+fn find_files<P: AsRef<Path> + ?Sized>(
+    config_path: &P,
+    f: impl Fn(&PathBuf) -> bool,
+) -> Result<Vec<PathBuf>> {
+    let mut entries = read_dir(&config_path)
+        .with_context(|| format!("reading directory {}", config_path.as_ref().display()))?
+        .filter_map(Result::ok)
+        .filter_map(|e| if f(&e.path()) { Some(e.path()) } else { None })
```

Comment:
Instead of passing `e.path()` to the closure, it'd be cleaner I think to pass `&e` directly. Then for the `hostkeys` case, you can do `e.file_name()` and for the LUKS keys, you can do `e.metadata()?.is_file()`.